include/zephyr/net/wifi_nm_wpa_supp.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_WIFI_NM_WPA_SUPP_H_
+#define ZEPHYR_INCLUDE_NET_WIFI_NM_WPA_SUPP_H_
+
+#include <zephyr/net/wifi_nm.h>
+
+union wifi_nm_supp_mon_event_data {
+	struct wifi_nm_supp_event_auth_reject {
+		int auth_type;
+		int auth_transaction;
+		int status_code;
+		uint8_t bssid[ETH_ALEN];
+	} auth_reject;
+
+	struct wifi_nm_supp_event_connected {
+		uint8_t bssid[ETH_ALEN];
+		char ssid[SSID_MAX_LEN];
+		int id;
+	} connected;
+
+	struct wifi_nm_supp_event_disconnected {
+		uint8_t bssid[ETH_ALEN];
+		int reason_code;
+		int locally_generated;
+	} disconnected;
+
+	struct wifi_nm_supp_event_assoc_reject {
+		int status_code;
+		int reason_code;
+	} assoc_reject;
+
+	struct wifi_nm_supp_event_temp_disabled {
+		int id;
+		char ssid[SSID_MAX_LEN];
+		unsigned int auth_failures;
+		unsigned int duration;
+		char reason_code[32];
+	} temp_disabled;
+
+	struct wifi_nm_supp_event_reenabled {
+		int id;
+		char ssid[SSID_MAX_LEN];
+	} reenabled;
+
+	struct wifi_nm_supp_event_bss_added {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_added;
+
+	struct wifi_nm_supp_event_bss_removed {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_removed;
+
+	struct wifi_nm_supp_event_network_added {
+		unsigned int id;
+	} network_added;
+
+	struct wifi_nm_supp_event_network_removed {
+		unsigned int id;
+	} network_removed;
+
+	char wifi_nm_supp_event_str[64];
```

Comment:
Please avoid using magic constants, replace 64 by a define that has a descriptive name.

Reply:
Done.